src/common/options.cc

```diff
@@ -3221,9 +3221,9 @@ std::vector<Option> get_global_options() {
     .set_default(100)
     .set_description("Approximate missing objects above which to force auth_log_shard to be primary temporarily"),
 
-    Option("osd_async_recovery_min_pg_log_entries", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    Option("osd_async_recovery_approx_missing_objects", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
```

Comment:
Not sure this name is appropriate, since this option now accounts for difference in length of logs plus missing objects.

Reply:
eh, what is your suggestion, then? @neha-ojha

Reply:
I see it as a cost of recovery, more so because now it seems to be dependent on more than one parameter.

Reply:
@neha-ojha 

What about:
```c++
Option("osd_async_recovery_min_cost", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
set_description("A mixture measure of number of current log entries difference and historical missing objects,  above which we switch to use asynchronous recovery when appropriate")
```

Reply:
sounds good to me

Reply:
@neha-ojha Repushed now!